### PR TITLE
Migrate simplify expressions

### PR DESCRIPTION
**Pull request by szymonm**

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/sql/planner/PlanOptimizers.java [hunk 1]
```diff
@@ -66,6 +66,7 @@
 import com.facebook.presto.sql.planner.iterative.rule.RemoveTrivialFilters;
```
Comment (kokosing) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/PlanOptimizers.java]: please sort commits before posting a PR, I expected this commit to be the last one, not the first.

File: presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/SimplifyExpressions.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+{
+    private final Metadata metadata;
+    private final SqlParser sqlParser;
+
+    private final Set<Rule<?>> rules = ImmutableSet.of(
+                    new ProjectExpressionRewrite(this::rewrite),
+                    new FilterExpressionRewrite(this::rewrite),
+                    new TableScanExpressionRewrite(this::rewrite),
+                    new JoinExpressionRewrite(this::rewrite),
+                    new ValuesExpressionRewrite(this::rewrite));
```
Comment (kokosing) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/SimplifyExpressions.java]: please write a comment which is node is missing here and please give short explanation why

**Comment by kokosing** [comment]
Also make sure that travis is passing.

File: presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/SimplifyExpressions.java [hunk 3]
```diff
@@ -81,6 +83,6 @@ public SimplifyExpressions(Metadata metadata, SqlParser sqlParser)
 
     private Expression rewrite(Expression expression, Rule.Context context)
     {
-        return rewrite(expression, context, metadata, sqlParser);
+        return rewrite(expression, context.getSession(), context.getSymbolAllocator(), metadata, sqlParser);
```
Comment (kokosing) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/SimplifyExpressions.java]: * can you please run it twice here, to make sure that subsequent execution does not rewrite anything. 
 * Also I think it would be valuable to extract some kind of `BaseExpressionRewriteTest` which you could then use for desugar rewrites. (it can be done later with desugar)

File: presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestSimplifyExpressions.java [hunk 4]
```diff
@@ -188,7 +139,7 @@ public Expression rewriteLogicalBinaryExpression(LogicalBinaryExpression node, V
         {
             List<Expression> predicates = extractPredicates(node.getType(), node).stream()
                     .map(p -> treeRewriter.rewrite(p, context))
-                    .sorted((p1, p2) -> p1.toString().compareTo(p2.toString()))
+                    .sorted(Comparator.comparing(Expression::toString))
```
Comment (kokosing) [inline presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestSimplifyExpressions.java]: unrelated?

Comment (szymonm) [reply presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestSimplifyExpressions.java]: It's just cleaning (with no semantic change) in the file I already change a lot. Not worth for another PR, IMO.
